### ISSUE TITLE
Focus pane based on click location when foregrounding

### ISSUE THREAD

**Issue by ianterrell** [warpdotdev/Warp#739]
### Describe the solution you'd like?

It would be nice if Warp focused on the pane clicked on when coming back from the background. Currently I have to click twice to select the pane I want to use.

I use a multilane workflow and this catches me quite a lot; I find myself typing into the wrong pane.

See comparison with iTerm2, even with focus follows mouse disabled:

https://user-images.githubusercontent.com/54963/153064916-d843f8c0-18a7-4e86-8278-d7dbc68955e6.mov

Combined with other issues, it makes me wonder if I'm in a minority using multiple panes in my terminal.
- https://github.com/warpdotdev/Warp/issues/304
- https://github.com/warpdotdev/Warp/issues/510
- https://github.com/warpdotdev/Warp/issues/699

**Comment by fwouts** [comment]
I'm still seeing this bug in v0.2022.04.01.01.37.stable_03 on MacOS 12.3.

@elviskahoro was it included in that release?

**Comment by elviskahoro** [comment]
I think we had to revert it. Thanks for commenting!

**Comment by elviskahoro** [comment]
> Often I click just to make sure the window is focused and I don't want to highlight/select anything.

**Comment by lobocv** [comment]
Would really love to see this feature. It drives me nuts when I change from my IDE to the terminal and start typing a command only to realize I need to click it again! It would be great if when you click on warp the first time, it will go to the cursor of the pane you clicked on. I think right now it focuses on the last output command and doesn't show the cursor prompt until I click again right near the text input

**Comment by jsharpe** [comment]
@elviskahoro given you had a fix for this can't this be reapplied in the latest versions? The original fix was made in March, its now August which suggests that this has fallen off your priority list?

**Comment by elviskahoro** [comment]
Good point. The original feature ended up causing another bug and so we removed it. After looking into the bug, we realized it be a much bigger engineering lift than anticipated. I'll reach back out and see.

**Comment by elviskahoro** [comment]
It's too big of an engineering lift at the moment. An so it's been deprioritized.

**Comment by korygorsky** [comment]
+1

Would love this update as well.

**Comment by Ken-vdE** [comment]
This is one of my biggest current issues. I can't count the amount of times i've ctrl-c'd the wrong running command because warp did not focus the pane I clicked. Very annoying.
Is there a ETA on this fix?

**Comment by dannyneira** [comment]
@Ken-vdE It is something we've begun work on, but no solid ETA on the fix. Thanks for your patience!

**Comment by ScopeSV** [comment]
I usually have multiple panes open, one might be vim, another one is running w/e I'm developing.
I have to manually click on the pane running vim if its not selected. This can also lead to quitting the program, as some shortcuts might be pressed. I'm running everything through the terminal as I really don't enjoy clicking around with the mouse that much. This makes an extra click step which is very prone to errors :)

**Comment by AshMcConnell** [comment]
Would love this to be fixed, it can be quite dangerous to type in the wrong window

**Comment by degecko** [comment]
Can you please provide this as an option if you can't make it the default bahaviour? It's really bothering in terms of UX. I'm constantly writing in the wrong terminal.

**Comment by dannyneira** [comment]
Edited to include Tabs as well as Panes since the behavior is the same when foregrounding.

**Comment by tylerstillwater** [comment]
Adding my two cents on this. This is the #1 disruptor of my daily workflow, and my biggest driver away from Warp. Warp is *so good* that I'm putting up with it for now, but I would be delighted to see this enhancement happen!

**Comment by cdcasey** [comment]
Absolute +1. Thanks for looking at the feedback, y'all!

**Comment by zachlloyd** [comment]
I'm working on a fix for this now.

here's the version for panes: https://share.getcloudapp.com/wbuLJlnp

just noticed this is also an issue for tabs - looking to see if i can fix that as well

**Comment by MrSpock** [comment]
Hi,
It's almost 3 years for this bug/feature request for option to disable.
This frustrating marking and changing color when clicking  terminal is only reason I'm not using warp as my daily driver.
You click terminal very often to confirm focus on its window. If focus is already there you end up with whole screen selected.

**Comment by meanderix** [comment]
I'm switching very frequently between windows and this is an absolutely dealbreaker. I will go back to Hyper and fzf.

**Comment by suecom** [comment]
Yes, exactly as stated (in my duplicate issue) - this is real pain (having to double click)

**Comment by persiyanov** [comment]
This is ridiculous. The most beautiful terminal app with the most annoying productivity bug I've seen in years.

Really hoping this gets fixed soon.

**Comment by dannyneira** [comment]
Hey folks, this has been fixed in the latest versions of Warp. Please update and let us know if any issues.
